forage-android/src/main/java/com/joinforage/forage/android/pos/ForageTerminalSDK.kt

```diff
@@ -0,0 +1,189 @@
+     * returns a detailed error response for proper handling.
+     *
+     * @throws ForageConfigNotSetException If the passed ForagePANEditText instance
+     * hasn't had its ForageConfig set via .setForageConfig().
+     */
+    suspend fun tokenizeEBTCard(
+        foragePanEditText: ForagePANEditText,
+        reusable: Boolean = true
+    ): ForageApiResponse<String> {
+        logger.i(
+            "[POS] Tokenizing Payment Method via UI PAN entry",
+            attributes = mapOf(
+                "reusable" to reusable
+            )
+        )
+
+        return forageSdk.tokenizeEBTCard(
+            TokenizeEBTCardParams(
+                foragePanEditText = foragePanEditText,
+                reusable = reusable
+            )
+        )
+    }
+
+    /**
+     * Tokenizes an EBT card using track 2 data from a magnetic card swipe.
+     *
+     * @param track2Data The track 2 data obtained from the magnetic stripe of the card.
+     * @param reusable Optional. Indicates whether the tokenized card can be
+     * reused for multiple transactions. Defaults to true.
+     *
+     * @return A ForageAPIResponse indicating the success or failure of the operation.
+     * On success, returns a [PaymentMethod](https://docs.joinforage.app/reference/create-payment-method)
+     * token which can be securely stored and used for subsequent transactions. On failure,
+     * returns a detailed error response for proper handling.
+     */
+    suspend fun tokenizeEBTCard(
+        track2Data: String,
+        reusable: Boolean = true
+    ): ForageApiResponse<String> {
+        logger.i(
+            "[POS] Tokenizing Payment Method via magnetic card swipe",
+            attributes = mapOf(
+                "reusable" to reusable
+            )
+        )
+
+        return ForageApiResponse.Success("TODO")
+    }
+
+    /**
+     * TODO: add comment here
+     */
+    suspend fun refundPosPayment(
+        params: RefundPosPaymentParams
+    ): ForageApiResponse<String> {
+        return ForageApiResponse.Success("TODO")
+    }
+
+    // ======= Same as online-only Forage SDK below =======
+
+    /**
+     * Checks the balance SNAP and EBT Cash balances of an EBT account via
+     * ForagePINEditText
+     *
+     * @param params The parameters required for tokenization, including
+     * reference to a ForagePINEditText and PaymentMethod ref
+     *
+     * @return A ForageAPIResponse indicating the success or failure of the operation.
+     * On success, returns an object with `snap` and `cash` fields, whose values
+     * indicate the balance of each tender as of now
+     *
+     * @throws ForageConfigNotSetException If the passed ForagePINEditText instance
+     * hasn't had its ForageConfig set via .setForageConfig().
+     */
+    override suspend fun checkBalance(
+        params: CheckBalanceParams
+    ): ForageApiResponse<String> {
+        return ForageApiResponse.Success("TODO")
```

Comment:
Not sure if this matters much, but you could return `return forageSdk.checkBalance(params)` like in the other functions.

Reply:
`checkBalance` will be different for the `ForageTerminalSDK` so the placeholder can be anything for now.

Reply:
Just chiming in here to make sure we're all on the same page

The signature for `checkBalance` is the same for `ForageSDK` and `ForageTerminalSDK`. However, in `ForageTerminalSDK`, we need to ferry the `pos_terminal_id` param to Payments API. This deception is also present with `ForageTerminalSDK.tokenizeCard`. Even though `ForageTerminalSDK.tokenizeCard` for manual PAN entry has the same signature as `ForageSDK.tokenizeCard` in online-only, the former needs to ferry the `pos_terminal_id` value to Payments API whereas the latter does not. This means the implementations will diverge.

---
#### (optional in case extra context is helpful)

[Here is the conversation](https://linear.app/joinforage/issue/FX-769/modify-checkbalancerepository-vgspincollector-btpincollector-to-pass#comment-0d5c200e) Tiffany clarifies that Payments API need to receive the `pos_terminal_id` param in the call to:
1. tokenize an EBT Card
2. checking a balance
3. issuing a refund.

Reply:
[Related LInear discussion](https://linear.app/joinforage/issue/FX-863/have-tokenizecardservice-support-manual-pan-entry-for-pos#comment-711299ff)

> However, in ForageTerminalSDK, we need to ferry the pos_terminal_id param to Payments API. This deception is also present with

Thank you for confirming @devinmorgan

Reply:
no, thank you!